devito/passes/iet/misc.py

```diff
@@ -225,6 +226,11 @@ def _(expr):
         return ()
 
 
+@_lower_macro_math.register(SafeInv)
+def _(expr):
+    return (('SAFEINV(a, b)', '(((a) < 1e-12 || (b) < 1e-12) ? (0.0F) : (1.0F / (a)))'),)
```

Comment:
the 1e-12 bounds should ideally be pulled from the SafeInv's attributes (default to 1e-12, but allow caller/user to pass a different threshold?)

Reply:
it's a safe threshold. But this is also only needed for single precision, double precision should probably default to standard div (not that we ever run double prec)

Reply:
yeah, it's just that like that it's a magic number buried deep down inside a compiler pass, so it should rather be pulled from `expr` itself (can do as a class attribute)

Reply:
done